src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -917,6 +940,31 @@ def _create_schedule(name):
         '''
         return KernelSchedule(name)
 
+    def generate_psyir(self, parse_tree):
+        '''Translate the supplied fparser2 parse_tree into PSyIR.
+
+        :param parse_tree: the supplied fparser2 parse tree.
+        :type parse_tree: :py:class:`fparser.two.Fortran2003.Program`
+
+        :returns: PSyIR representation of the supplied fparser2 parse_tree.
+        :rtype: :py:class:`psyclone.psyir.nodes.Container` or \
+            :py:class:`psyclone.psyir.nodes.Routine`
+
+        :raises GenerationError: if the root of the supplied fparser2 \
+            parse tree is not a Program.
+
+        '''
+        if not isinstance(parse_tree, Fortran2003.Program):
+            raise GenerationError(
+                "The Fparser2Reader generate_psyir method expects root the "
```

Comment:
Is this error message incomplete?

Reply:
Oops, bad English. It should be "expects the root of the". Modified this and the associated test.